variables.tf

```diff
@@ -0,0 +1,136 @@
+variable "location" {
+  description = "Location of resource(s)."
+  type        = string
+}
+
+variable "contributors" {
+  description = "List of service principal ID to have Contributor role for the PostgreSQL Flexible Server."
+  type = list(string)
+  default = []
+}
+
+variable "users_on_keyvault" {
+  description = "List of object ID (user, service principal or security group in the AD tenant) to have predifined user-like required access to kv."
+  type = list(string)
+  default = []
+}
+
+variable "terraformers_on_keyvault" {
+  description = "List of object ID (user, service principal or security group in the AD tenant) to have predifined terraform-like required access to kv."
+  type = list(string)
+  default = []
+}
+
+variable "pg_version" {
+  description = "Postgresql engine version"
+  type        = string
+  default     = "13"
+}
+
+variable "administrator_login" {
+  description = "Postgresql admin name"
+  type        = string
+  default     = "psqladmin"
+
+}
+
+variable "sku_name" {
+  description = "Postgresql vm size"
+  type        = string
+  default     = "GP_Standard_D2s_v3"
+}
+
+variable "storage_mb" {
+  description = "Postgresql size storage in megabytes"
+  type        = number
+  default     = 32768
+}
+
+variable "resource_group_name" {
+  description = "Resource group name."
+  type        = string
+}
+
+variable "tenant_id" {
+  description = "Tenant ID."
+  type        = string
+}
+
+variable "virtual_network_name" {
+  description = "Virtual Network Name where Subnet will be created."
+  type        = string
+}
+
+variable "virtual_network_id" {
+  description = "Virtual Network ID where Subnet will be created."
+  type        = string
+}
+
+variable "virtual_network_pipeline_id" {
+  description = "Virtual Network ID where Subnet will be created."
+  type        = string
+}
+
+variable "name" {
+  description = "Name for the PostgreSQL Flexible Server."
+  type        = string
+}
+
+variable "subnet_address_prefixes" {
+  description = "Address prefixes of Subnet to be created."
+  type        = list(string)
+}
+
+variable "backup_retention_days" {
+  description = "Backup retention days for the PostgreSQL Flexible Server."
+  type        = number
+  default     = 35
+}
+
+variable "maintenance_window" {
+  description = "Cron maintenance window (default sunday midnight)."
+  type = object({
+    day_of_week  = number
+    start_hour   = number
+    start_minute = number
+  })
+  default = {
+    day_of_week  = 0
+    start_hour   = 0
+    start_minute = 0
+  }
+}
+
+variable "private_dns_zone_name" {
+  description = ""
```

Comment:
"The ID of the private DNS zone to create the PostgreSQL Flexible Server. Changing this forces a new PostgreSQL Flexible Server to be created."